sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/graph/AadGraphClient.java

```diff
@@ -176,7 +176,11 @@ public IAuthenticationResult acquireTokenForGraphApi(String idToken, String tena
             scopes.add(MICROSOFT_GRAPH_SCOPE);
             final OnBehalfOfParameters onBehalfOfParameters = OnBehalfOfParameters.builder(scopes, assertion).build();
             result = application.acquireToken(onBehalfOfParameters).get();
-        } catch (ExecutionException | InterruptedException | MalformedURLException e) {
+        } catch (InterruptedException e) {
+            LOGGER.warn("Interrupted during acquiring token for graph API!", e);
+            // Restore interrupted state...
+            Thread.currentThread().interrupt();
```

Comment:
shall we add a test case to verify this?

Reply:
InterruptedExceptions should never be ignored in the code, and simply logging the exception counts in this case as "ignoring". The throwing of the InterruptedException clears the interrupted state of the Thread, so if the exception is not handled properly the information that the thread was interrupted will be lost. Instead, InterruptedExceptions should either be rethrown - immediately or after cleaning up the method’s state - or the thread should be re-interrupted by calling Thread.interrupt() even if this is supposed to be a single-threaded application. Any other course of action risks delaying thread shutdown and loses the information that the thread was interrupted - probably without finishing its task.